### PR TITLE
feat: show newest agreement

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- display the most recent agreement, but only to the creator, even if staking threshold is not met
- fixed an issue where the entire list was reloaded on add. Now, only the first item is added to the list without refreshing the entire content

image

## Testing

* Go to agreements page.
* Create a agreements using Reputation decision method. Some passing 10% threshold and some not.
* After adding new agreements, check if it's visible without staking and if whole list is not reloading when adding new agreements.

Resolves https://github.com/JoinColony/colonyCDapp/issues/3240

### CODE REVIEW

**Review by arrenv — CHANGES_REQUESTED** [review]
@CzarekDryl Great, work. Thank you adding this so quick. As mentioned, I thought this may be a good one to test out slightly to see how it worked. So, based on that, I have a request.

Currently, it very conveniently shows you your recent Agreement that is below the threshold, but, it only shows you 1, and as soon as you create a new one it disappears. Additionally, it is not really clear what it means, so the requests are.

1. Could we show all the users created agreements, even the ones below the threshold, in the correct order throughout the feed sorted by date.

2. Can we add a tooltip on the "Below threshold" pill that includes the text: "This agreement is below the 10% staking threshold and will not be visible to others by default."


3. Lastly, unrelated to this issue slightly, but is there are way to stop the draft agreement card flashing in when creating a new agreement?

[draft-flashing.webm](https://github.com/user-attachments/assets/17d3049f-36bf-4fff-9311-3b135796f592)

File: src/components/v5/common/Pills/PillsBase.tsx [hunk 1]
```diff
@@ -32,7 +33,7 @@ const PillsBase: FC<PropsWithChildren<PillsProps>> = ({
       {Icon && (
         <span className="flex shrink-0">
           <Icon
-            size={pillSize === 'medium' ? 14 : 12}
+            size={iconSize || (pillSize === 'medium' ? 14 : 12)}
```
Comment (mmioana) [inline src/components/v5/common/Pills/PillsBase.tsx:36]: Is there anything speaking against using the `small` value for the `pillSize` property?

File: src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx [hunk 2]
```diff
@@ -75,7 +76,18 @@ const AgreementCard: FC<AgreementCardProps> = ({ transactionId }) => {
...
-            {motionState && <MotionStateBadge state={motionState} />}
+            <div className="flex items-center gap-1">
+              {motionState && <MotionStateBadge state={motionState} />}
+              {!showInActionsList && (
+                <PillsBase
+                  icon={WarningCircle}
+                  iconSize={12}
+                  className="bg-warning-100 text-warning-400"
+                >
+                  Below threshold
```
Comment (mmioana) [inline src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx:87]: Could we please add `Below threshold` to a translation message descriptor?

```
const displayName = 'frame.v5.pages.AgreementsPage.partials.AgreementCard';
const MSG = defineMessages({
  belowThresholdStatus: {
    id: `${displayName}.belowThresholdStatus`,
    defaultMessage: 'Below threshold',
  },
});
```

File: src/components/frame/v5/pages/AgreementsPage/AgreementsPage.tsx [hunk 3]
```diff
@@ -52,6 +52,22 @@ const AgreementsPage: FC = () => {
...
   );
 
+  const passedAgreements = searchedAgreements.filter(
+    (agreement) => agreement.showInActionsList,
+  );
+  const newestAgreement = searchedAgreements.sort((a, b) => {
+    const aDate = new Date(a?.decisionData?.createdAt || '');
+    const bDate = new Date(b?.decisionData?.createdAt || '');
+    return bDate.getTime() - aDate.getTime();
+  })[0];
```
Comment (mmioana) [inline src/components/frame/v5/pages/AgreementsPage/AgreementsPage.tsx:62]: Can we extract this into an utility function?

**Review by mmioana — CHANGES_REQUESTED** [review]
Hey @CzarekDryl nice work so far! 👏 

Beside the changes requested by Arren, I left a few comments regarding the code, if you could please tackle them 🙏

File: src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx (hunk 2 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/AgreementsPage/partials/AgreementCard/AgreementCard.tsx]: Added as translation :)

**Comment by CzarekDryl** [comment]
@arrenv Ready to be checked with new requirements
image

**Review by arrenv — CHANGES_REQUESTED** [review]
@CzarekDryl Great work, thank you for adding this. So much better.


1. There is a key issue in the action panel for agreements.
image

2. The order seems to prioritize ones that are below threshold. The order should be chronological. As you can see in my numbered agreements.
image


Everything else looks good

They all remain and correctly have the Below threshold pill.
image

Tooltip works well:
image

**Review by mmioana — CHANGES_REQUESTED** [review]
Nice work @CzarekDryl on this one and thanks for tackling all comments 🙌 💯 

The agreements seem to be shown in the correct order by date
Screenshot 2024-10-23 at 14 14 43

The tooltip is showing the correct copy
Screenshot 2024-10-23 at 14 14 51

And the error regarding the unique key is no longer shown in the console - it's only the error related to uncontrolled inputs but this should be caused by the action sidebar and don't think it should be treated in this PR
Screenshot 2024-10-23 at 14 15 42

Left a few very small refactoring comments, if you mind to tackle them 🥇

**Comment by CzarekDryl** [comment]
@mmioana Done :)

**Review by mmioana — APPROVED** [review]
All good from my side 🙌  thanks for doing these changes so fast @CzarekDryl 🥇 💯 

Screenshot 2024-10-24 at 10 18 29
Screenshot 2024-10-24 at 10 19 05

**Review by arrenv — APPROVED** [review]
@CzarekDryl Nice one, thank you!!! Looks and works great!

image

**Pull request merged**